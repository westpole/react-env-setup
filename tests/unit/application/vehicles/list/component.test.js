import toJson from 'enzyme-to-json';

import Component from '../../../../../src/scripts/application/vehicles/list/component.jsx';
import { mockComponent } from '../../../../tools/mock-component.jsx';

describe('Application/Vehicles/List/Component', () => {
  it('should match saved snapshot', () => {
    const wrapper = mockComponent({
      type: 'shallow',
      scope: Component,
      props: {
        vehicles: [],
        error: {},
      },
    });

    expect(toJson(wrapper)).toMatchSnapshot();
  });

  it('should generate a list of data', () => {
    const wrapper = mockComponent({
      type: 'shallow',
      scope: Component,
      props: {
        vehicles: [
          {
            name: 'Colombo',
            cargo_capacity: '10',
            passengers: '1',
            vehicle_class: 'suv',
            cost_in_credits: '1234',
          },
        ],
        error: {},
      },
    });

    const list = [];

    expect(wrapper.find('.vehicle-entity').length).toEqual(1);
    expect(wrapper.find('div[data-type]').length).toEqual(5);

    wrapper.find('div[data-type]').forEach((node) => {
      list.push(node.text());
    });

    expect(list[0]).toEqual('Colombo');
    expect(list[1]).toEqual('10');
    expect(list[2]).toEqual('1');
    expect(list[3]).toEqual('suv');
    expect(list[4]).toEqual('1234');
  });

  it('should display error message', () => {
    const wrapper = mockComponent({
      type: 'shallow',
      scope: Component,
      props: {
        vehicles: [],
        error: {
          message: 'test error',
        },
      },
    });

    expect(wrapper.find('.error').length).toEqual(1);
  });
});
