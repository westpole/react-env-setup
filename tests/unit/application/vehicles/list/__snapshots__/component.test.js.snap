// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Application/Vehicles/List/Component should match saved snapshot 1`] = `
<article
  className="table"
>
  <section
    className="header"
  >
    <div
      className="cell"
      key="1"
    >
      Name
    </div>
    <div
      className="cell"
      key="1"
    >
      Cargo capacity
    </div>
    <div
      className="cell"
      key="1"
    >
      Passengers
    </div>
    <div
      className="cell"
      key="1"
    >
      Vehicle class
    </div>
    <div
      className="cell"
      key="1"
    >
      Cost in credits
    </div>
  </section>
</article>
`;
